profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry_mock/src/Mock/PoetryMock.php

```diff
@@ -461,6 +489,40 @@ public function refuseJob($file_id, $tjiid) {
   }
 
   /**
+   * Sends given status with messages to Drupal.
+   *
+   * @param int $job_id
+   *   TMGMT Poetry Job ID.
+   * @param string $status_code
+   *   The status code.
+   * @param string $request_status_msg
+   *   The request status message.
+   * @param string $demande_status_msg
+   *   The 'demande' status message.
+   */
+  public function sendStatus($job_id, $status_code, $request_status_msg, $demande_status_msg) {
+    $job = tmgmt_job_load($job_id);
+    $lg_code = drupal_strtoupper($job->getTranslator()->mapToRemoteLanguage($job->target_language));
+    $status_code = drupal_strtoupper($status_code);
+    $demande_id = self::prepareDemandeIdArray($job->reference);
+
+    // Prepare responses array.
+    $data = self::prepareSendStatusData($demande_id, $status_code, $request_status_msg, $demande_status_msg, $lg_code);
+    $message = theme('poetry_send_status', $data);
+    $this->sendRequestToDrupal($message);
+    $entity = self::getEntityDetailsByDemandeId($demande_id);
+    $msg = t('The status request was sent. !link.', [
+      '!link' => l(
+        t('Check the translation page'),
+        $entity['entity_type'] . '/' . $entity['entity_id'] . '/translate'
+      ),
+    ]);
+    drupal_set_message($msg, 'status');
+
+    drupal_goto();
```

Comment:
Yes it is needed to redirect.

See the $action_list item keyed 'send_ong_status', in _poetry_mock_get_data_row_action_links().